vms/evm/subscriber.go

```diff
@@ -121,11 +129,31 @@ func (s *subscriber) processBlockRange(
 			)
 			return err
 		}
-		s.headers <- header
+		blocks, err := relayerTypes.LogsToBlocks(logs)
```

Comment:
```suggestion
		blocksWithICMMessages, err := relayerTypes.LogsToBlocks(logs)
```
To clarify that this is not expected to be a contiguous list